Add initial contributing doc

twpayne opened the pull request:
It came up in yesterday's Cilium meeting that this project does not have a contributing doc. This PR adds an initial one and also tidies up the README a little.

Fixes https://github.com/cilium/ebpf/issues/184

cc @joestringer

tklauser commented:
Also see #184 and https://cilium.slack.com/archives/CKC55JL8L/p1607371973036400 for discussion.

lmb commented:
Could you add a commit that only renames readme.md? Seems like it's throwing off the diff somehow, I can't tell what has changed.

The contribution guidelines sgtm largely, I think we should include that new features have to be accompanied by tests, and that it's best to drop into the slack channel before starting work on a big feature to discuss the design. I'm not super fussed about commit messages, the more detail the better, etc.

twpayne commented:
Done:
* Added separate rename commit.
* Changes to `README.md` are now in f66f792ff02a3952b46d2ddb414766e045be5d92. It's basically just line wrapping, consistent punctuation, consistent Markdown styling, adding a License section, and adding a `#` before the Slack channel name.
* Updated `CONTRIBUTING.md` to include what @lmb requested.

joestringer commented on CONTRUBUTING.md:
```diff
@@ -0,0 +1,17 @@
+
+## Running the tests
+
+Many of the tests require priviliges to set resource limits and load eBPF code.
```
```suggestion
Many of the tests require privileges to set resource limits and load eBPF code.
```

joestringer commented on CONTRUBUTING.md:
```diff
@@ -0,0 +1,17 @@
+the form of pull requests and issues reporting bugs or suggesting new features
+are welcome.
+
+New features must be acompanied by tests. Before starting work on any large
```
```suggestion
New features must be accompanied by tests. Before starting work on any large
```

joestringer reviewed (changes requested):
Couple of minor spelling nits below.

Do we want to mention something about describing the problems as well? eg:

```
When submitting pull requests, consider writing details about what problem you are solving and why the proposed approach solves that problem in commit messages and/or pull request description to help future library users and maintainers to reason about the proposed changes.
```

twpayne commented:
Thanks for the feedback, PR updated.

twpayne commented:
Gentle ping: can this be merged now?

thaJeztah commented:
This may be specific to people's situation, but leaving it here in case useful to others; if you're running into issues because git doesn't like the case-only filename change (`readme.md` -> `README.md`);

- if you're on a case-insensitive (but case-preserving) filesystem (macOS, Windows): configure git to not ignore case-only changes;

     ```bash
    git config core.ignorecase false
    ```
- if your CI doesn't like this (for example, runc CI failed on the rename https://github.com/opencontainers/runc/pull/2875); adding a `.gitconfig` to your github repository to make the change configuration can help;

    ```config
    [core]
     	ignorecase = false
    ```

thaJeztah commented:
Hmm.. that config can actually be a bit of a pain as well, because there's no content changes in the readme, git still gets confused when switching branches 😓 

```bash
git checkout master
error: The following untracked working tree files would be overwritten by checkout:
	vendor/github.com/cilium/ebpf/readme.md
Please move or remove them before you switch branches.
Aborting
```
